Various cleanups

zx2c4 opened the pull request:
Since updating the cache the first time or after not syncing for a while takes quite a bit of CPU, we might as well opportunistically utilize the rsync cache by selectively rsyncing from it, and then running the local caching operation after.

We also cleanup the output and simplify the git pulling using git's -C option.

Also, it's better to use api.g.o instead of gitweb.

zx2c4 committed:
sync_cache: Use the rsync cache first

Since updating the cache the first time or after not syncing for a while
takes quite a bit of CPU, we might as well opportunistically utilize the
rsync cache by selectively rsyncing from it, and then running the local
caching operation after.

Signed-off-by: Jason A. Donenfeld <Jason@zx2c4.com>

hasufell commented on etc/portage/postsync.d/sync_cache:
```diff
@@ -5,5 +5,6 @@ source /etc/init.d/functions.sh
 source /etc/portage/util-functions.sh
```
we should source `/etc/portage/check-portdir.sh` here and then...

zx2c4 replied:
We do. It's in the line just above. Expand the diff context.

hasufell replied:
oh, nasty github

hasufell commented on etc/portage/postsync.d/sync_cache:
```diff
@@ -5,5 +5,6 @@ source /etc/init.d/functions.sh
 source /etc/portage/util-functions.sh
 
 einfo "syncing metadata cache"
+rsync -avP rsync://rsync.gentoo.org/gentoo-portage/metadata/md5-cache/ /usr/portage/metadata/md5-cache/
```
use `${PORTDIR}` here instead of hardcoded `/usr/portage`

zx2c4 committed:
Cleanup output and scripting

Using ebegin/eend makes things a bit cleaner and Gentoo-ish, and we can
also clean up the scripts in general by using git's -C option.

Signed-off-by: Jason A. Donenfeld <Jason@zx2c4.com>

zx2c4 committed:
Use proper URL for herds

zx2c4 committed:
Don't hardcode 8 jobs. Use nproc.

hasufell commented:
I'll test these things shortly. Unfortunately I'm a bit busy right now.

zx2c4 commented:
The commits are perfect. You can safely just merge the pull request. :-P

zx2c4 committed:
This isn't as useful as it seems

hasufell commented on etc/portage/postsync.d/sync_cache:
```diff
@@ -4,6 +4,7 @@ source /etc/portage/check-portdir.sh
-egencache --jobs=8 --repo=gentoo --update --update-use-local-desc || die "egencache failed!"
-einfo "done syncing metadata cache"
+ebegin "Syncing metadata cache"
+rsync -avP rsync://rsync.gentoo.org/gentoo-portage/metadata/md5-cache/ "${PORTDIR}"/metadata/md5-cache/
```
I find this way too verbose, hiding all my other previous messages, let's do `-q`.
